packages/adapters-library/src/adapters/prices-v2/products/usd/pricesV2UsdAdapter.ts

```diff
@@ -1,41 +1,78 @@
 import { getAddress } from 'ethers'
 import { AdaptersController } from '../../../../core/adaptersController'
 import { Chain } from '../../../../core/constants/chains'
-import { NotImplementedError } from '../../../../core/errors/errors'
+
 import { CustomJsonRpcProvider } from '../../../../core/provider/CustomJsonRpcProvider'
-import { getTokenMetadata } from '../../../../core/utils/getTokenMetadata'
+
 import { nativeTokenAddresses } from '../../../../core/utils/nativeTokens'
 import { Helpers } from '../../../../scripts/helpers'
+
 import {
-  IProtocolAdapter,
-  ProtocolToken,
-} from '../../../../types/IProtocolAdapter'
-import {
-  GetEventsInput,
-  GetPositionsInput,
-  GetTotalValueLockedInput,
-  MovementsByBlock,
-  PositionType,
-  ProtocolAdapterParams,
-  ProtocolDetails,
-  ProtocolPosition,
-  ProtocolTokenTvl,
+  AdapterSettings,
   TokenType,
   UnwrapExchangeRate,
-  UnwrapInput,
 } from '../../../../types/adapter'
-import { Protocol } from '../../../protocols'
+
+import { Erc20Metadata } from '../../../../types/erc20Metadata'
 import { ChainLink__factory, OneInchOracle__factory } from '../../contracts'
 import { priceAdapterConfig } from './priceV2Config'
 
 export const USD = 'USD'
 
+export interface SpotPrice {
+  allTimeHigh?: number
+  allTimeLow?: number
+  circulatingSupply?: number
+  dilutedMarketCap?: number
+  high1d?: number
+  id?: string
+  low1d?: number
+  marketCap?: number
+  marketCapPercentChange1d?: number
+  price: number
+  priceChange1d?: number
+  pricePercentChange1d?: number
+  pricePercentChange1h?: number
+  pricePercentChange1y?: number
+  pricePercentChange7d?: number
+  pricePercentChange14d?: number
+  pricePercentChange30d?: number
+  pricePercentChange200d?: number
+  totalVolume?: number
+}
+
+export interface GetSpotPriceByAddressInput {
+  tokenAddresses: string
+  vsCurrency?: string
+  chainId: number
+}
+
+export type IPricesInput = {
+  blockNumber?: number
+  tokenMetadata: Erc20Metadata
+}
+export interface IPricesAdapter {
```

Comment:
Very glad you went for a different interface. We shouldn't be afraid of doing it when it feels too forced and it has drawbacks (such as not having the token metadata).